frontend/src/store/slices/searchSlice.ts

```diff
@@ -0,0 +1,67 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import type { RootState } from "..";
+
+export interface SearchResult {
+	type: "song" | "artist" | "playlist";
```

Comment:
we have a fixed amount of possible types here, we should also document that in backend.py using `Literal["song", "artist", "playlist"]`

Reply:
Got it!